.github/actions/k-bench/action.yml

```
@@ -0,0 +1,105 @@
+name: kbench
+description: "Run K-Bench"
+inputs:
+  githubReadRepos:
+    description: "GitHub authorization token to read repositories"
+    required: true
+  cloudProvider:
+    description: "Which cloud provider to use."
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup go
+      uses: actions/setup-go@84cbf8094393cdc5fe1fe1671ff2647332956b1a
+      with:
+        go-version: ">=1.18.0"
+    - name: Setup python
+      uses: actions/setup-python@c4e89fac7e8767b327bbad6cb4d859eda999cf08
+      with:
+        python-version: "3.10"
+    - name: Retrieve patched K-Bench
+      uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+      with:
+        fetch-depth: 0
+        token: ${{ inputs.githubReadRepos }}
+        repository: "edgelesssys/k-bench"
+        path: k-bench
+    - name: Install patched K-Bench
+      working-directory: "k-bench"
+      run: ./install.sh
+      shell: bash
+    - name: Install Constellation GCP CSI driver and create storage class
+      if: ${{ inputs.cloudProvider == 'gcp' }}
+      shell: bash
+      run: |
+        KUBECONFIG=$PWD/constellation-admin.conf kubectl apply -k github.com/edgelesssys/constellation-gcp-compute-persistent-disk-csi-driver/deploy/kubernetes/overlays/edgeless/latest
+        KUBECONFIG=$PWD/constellation-admin.conf kubectl wait -n kube-system deployments csi-gce-pd-controller --for condition=available
+        KUBECONFIG=$PWD/constellation-admin.conf kubectl apply -f .github/actions/k-bench/gcp_sc.yml
```

```suggestion
      run: |
        kubectl apply -k github.com/edgelesssys/constellation-gcp-compute-persistent-disk-csi-driver/deploy/kubernetes/overlays/edgeless/latest
        kubectl wait -n kube-system deployments csi-gce-pd-controller --for condition=available
        kubectl apply -f .github/actions/k-bench/gcp_sc.yml
      env:
            KUBECONFIG: '{{ github.workspace }}/constellation-admin.conf'
```

Could we set `KUBECONFIG` as an environment variable to remove the repetition? 
GitHub action also provides the [github context](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context) where the `workspace` path is available, which would allow us to set the environment for the step. 
(I did not test the suggestion)